File: PathoFroh/jpametamodel/com/patho/main/model/AssociatedContactNotification_.java
package com.patho.main.model;

import com.patho.main.model.AssociatedContactNotification.NotificationTyp;
import java.util.Date;
import javax.annotation.Generated;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(AssociatedContactNotification.class)
public abstract class AssociatedContactNotification_ {

	public static volatile SingularAttribute<AssociatedContactNotification, Boolean> renewed;
	public static volatile SingularAttribute<AssociatedContactNotification, NotificationTyp> notificationTyp;
	public static volatile SingularAttribute<AssociatedContactNotification, AssociatedContact> contact;
	public static volatile SingularAttribute<AssociatedContactNotification, Boolean> manuallyAdded;
	public static volatile SingularAttribute<AssociatedContactNotification, Boolean> active;
	public static volatile SingularAttribute<AssociatedContactNotification, String> contactAddress;
	public static volatile SingularAttribute<AssociatedContactNotification, Long> id;
	public static volatile SingularAttribute<AssociatedContactNotification, Boolean> failed;
	public static volatile SingularAttribute<AssociatedContactNotification, Date> dateOfAction;
	public static volatile SingularAttribute<AssociatedContactNotification, Boolean> performed;
	public static volatile SingularAttribute<AssociatedContactNotification, String> commentary;

}

File: PathoFroh/src/main/kotlin/com/patho/main/model/patient/notification/AssociatedContactNotification.kt
package com.patho.main.model.patient.notification

import com.patho.main.model.AbstractPersistable
import com.patho.main.model.AssociatedContact
import com.patho.main.model.interfaces.ID
import org.hibernate.annotations.SelectBeforeUpdate
import org.hibernate.envers.Audited
import java.util.*
import javax.persistence.*

/**
 *
 */
@Entity
@SequenceGenerator(name = "associatedcontactnotification_sequencegenerator", sequenceName = "associatedcontactnotification_sequence")
@Audited
@SelectBeforeUpdate(true)
open class AssociatedContactNotification : AbstractPersistable(), ID {

    @Id
    @GeneratedValue(generator = "associatedcontactnotification_sequencegenerator")
    @Column(unique = true, nullable = false)
    override var id: Long = 0

    @ManyToOne(targetEntity = AssociatedContact::class, fetch = FetchType.LAZY)
    open var contact: AssociatedContact? = null

    @Enumerated(EnumType.STRING)
    open var notificationTyp: NotificationTyp? = null

    @Column
    open var active: Boolean = false

    @Column
    open var performed: Boolean = false

    @Column
    open var failed: Boolean = false

    @Column
    open var renewed: Boolean = false

    @Column(columnDefinition = "VARCHAR")
    open var commentary: String? = null

    @Temporal(TemporalType.DATE)
    open var dateOfAction: Date? = null

    @Column
    open var manuallyAdded: Boolean = false

    @Column(columnDefinition = "VARCHAR")
    open var contactAddress: String? = null

    enum class NotificationTyp {
        EMAIL, FAX, PHONE, LETTER, PRINT
    }
}